File: src/main/java/com/thunderbear06/ai/modules/AndroidModules.java
package com.thunderbear06.ai.modules;

import com.thunderbear06.ai.AndroidBrain;
import com.thunderbear06.entity.android.BaseAndroidEntity;

public class AndroidModules {
    public final NavigationModule navigationModule;
    public final MiningModule miningModule;
    public final SensorModule sensorModule;
    public final InteractionModule interactionModule;

    public AndroidModules(BaseAndroidEntity android, AndroidBrain brain) {
        navigationModule = new NavigationModule(android, brain);
        miningModule = new MiningModule(android, brain);
        sensorModule = new SensorModule(android, brain, android.getEntitySearchRadius(), android.getBlockSearchRadius());
        interactionModule = new InteractionModule(android, brain);
    }
}

File: src/main/kotlin/com/thunderbear06/ai/modules/AndroidModules.kt
package com.thunderbear06.ai.modules

import com.thunderbear06.ai.AndroidBrain
import com.thunderbear06.entity.android.BaseAndroidEntity

class AndroidModules(android: BaseAndroidEntity, brain: AndroidBrain) {
    @JvmField val navigationModule: NavigationModule = NavigationModule(android, brain)
    @JvmField val miningModule: MiningModule = MiningModule(android, brain)
    @JvmField val sensorModule: SensorModule = SensorModule(android, brain, android.entitySearchRadius, android.blockSearchRadius)
    @JvmField val interactionModule: InteractionModule = InteractionModule(android, brain)
}
